src/python_testing/TC_CC_2_1.py

```diff
@@ -0,0 +1,400 @@
+            TestStep(
+                34, 'TH reads Primary4X attribute from DUT Verify that the DUT response contains an uint16 [Min:0 Max:0xfeff] if NumberOfPrimaries is 4 or more'),
+            TestStep(
+                35, 'TH reads Primary4Y attribute from DUT and Verify that the DUT response contains an uint16 [Min:0 Max:0xfeff] if NumberOfPrimaries is 4 or more'),
+            TestStep(36, 'TH reads Primary4Intensity attribute from DUT and Verify that the DUT response contains an uint8 if NumberOfPrimaries is 4 or more'),
+            TestStep(
+                37, 'TH reads Primary5X attribute from DUT and Verify that the DUT response contains an uint16 [Min:0 Max:0xfeff] if NumberOfPrimaries is 5 or more'),
+            TestStep(
+                38, 'TH reads Primary5Y attribute from DUT and Verify that the DUT response contains an uint16 [Min:0 Max:0xfeff] if NumberOfPrimaries is 5 or more'),
+            TestStep(39, 'TH reads Primary5Intensity attribute from DUT and Verify that the DUT response contains an uint8 if NumberOfPrimaries is 5 or more'),
+            TestStep(
+                40, 'TH reads Primary6X attribute from DUT and Verify that the DUT response contains an uint16 [Min:0 Max:0xfeff] if NumberOfPrimaries is 6 or more'),
+            TestStep(
+                41, 'TH reads Primary6Y attribute from DUT and Verify that the DUT response contains an uint16 [Min:0 Max:0xfeff] if NumberOfPrimaries is 6 or more'),
+            TestStep(42, 'TH reads Primary6Intensity attribute from DUT and Verify that the DUT response contains an uint8 if NumberOfPrimaries is 6 or more'),
+            TestStep(43, 'TH reads from the DUT the (0x0030) WhitePointX attribute'),
+            TestStep(44, 'TH reads from the DUT the (0x0031) WhitePointY attribute'),
+            TestStep(45, 'TH reads from the DUT the (0x0032) ColorPointRX attribute'),
+            TestStep(46, 'TH reads from the DUT the (0x0033) ColorPointRY attribute'),
+            TestStep(47, 'TH reads from the DUT the (0x0034) ColorPointRIntensity attribute'),
+            TestStep(48, 'TH reads from the DUT the (0x0036) ColorPointGX attribute'),
+            TestStep(49, 'TH reads from the DUT the (0x0037) ColorPointGY attribute'),
+            TestStep(50, 'TH reads from the DUT the (0x0038) ColorPointGIntensity attribute'),
+            TestStep(51, 'TH reads from the DUT the (0x003a) ColorPointBX attribute'),
+            TestStep(52, 'TH reads from the DUT the (0x003b) ColorPointBY attribute'),
+            TestStep(53, 'TH reads from the DUT the (0x003c) ColorPointBIntensity attribute')
+        ]
+
+        return steps
+
+    async def verify_primary_index(self, primary_index: int) -> bool:
+        # Read all PrimaryN<X,Y,Intensity> attributes available in the cluster
+        instance_attribute_names = [attr for attr in self.attributes.__dict__.keys(
+        ) if not attr.startswith("__") and attr.startswith('Primary')]
```

Comment:
```suggestion
        ) if not attr.startswith("_") and attr.startswith('Primary')]
```

In fact, why not just use `if attr.startswith('Primary')` ?